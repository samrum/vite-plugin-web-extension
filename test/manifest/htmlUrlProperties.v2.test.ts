import { getResourceDir, runManifestV2Test } from "./manifestTestUtils";

const resourceDir = getResourceDir("htmlUrlProperties");

runManifestV2Test("htmlUrlProperties", () => ({
  browser_action: {
    default_popup: `${resourceDir}/popup.html#hashValue`,
  },
  devtools_page: `${resourceDir}/devtools.html?query=1`,
  options_ui: {
    page: `${resourceDir}/options.html?query=1#hashValue`,
  },
}));
