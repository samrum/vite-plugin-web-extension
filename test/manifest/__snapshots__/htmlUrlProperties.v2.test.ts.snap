// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`htmlUrlProperties - Manifest V2 1`] = `
[
  {
    "code": "function log(message) {
  console.log(message);
}
export {
  log as l
};
",
    "dynamicImports": [],
    "exports": [
      "l",
    ],
    "facadeModuleId": null,
    "fileName": "assets/log.js",
    "implicitlyLoadedBefore": [],
    "importedBindings": {},
    "imports": [],
    "isDynamicEntry": false,
    "isEntry": false,
    "isImplicitEntry": false,
    "map": null,
    "modules": {
      "vite-plugin-web-extension/test/manifest/resources/shared/log.js": {
        "code": "function log(message) {
  console.log(message);
}",
        "originalLength": 65,
        "removedExports": [],
        "renderedExports": [
          "default",
        ],
        "renderedLength": 49,
      },
    },
    "name": "log",
    "referencedFiles": [],
    "type": "chunk",
    "viteMetadata": {
      "importedAssets": Set {},
      "importedCss": Set {},
    },
  },
  {
    "code": "import { l as log } from \\"../../../../log.js\\";
log(\\"devtools\\");
",
    "dynamicImports": [],
    "exports": [],
    "facadeModuleId": "vite-plugin-web-extension/test/manifest/resources/htmlUrlProperties/devtools.html",
    "fileName": "assets/test/manifest/resources/htmlUrlProperties/devtools.js",
    "implicitlyLoadedBefore": [],
    "importedBindings": {
      "assets/log.js": [
        "l",
      ],
    },
    "imports": [
      "assets/log.js",
    ],
    "isDynamicEntry": false,
    "isEntry": true,
    "isImplicitEntry": false,
    "map": null,
    "modules": {
      "vite-plugin-web-extension/test/manifest/resources/htmlUrlProperties/devtools.html": {
        "code": "",
        "originalLength": 227,
        "removedExports": [],
        "renderedExports": [],
        "renderedLength": 0,
      },
      "vite-plugin-web-extension/test/manifest/resources/htmlUrlProperties/devtools.js": {
        "code": "log(\\"devtools\\");",
        "originalLength": 53,
        "removedExports": [],
        "renderedExports": [],
        "renderedLength": 16,
      },
    },
    "name": "test/manifest/resources/htmlUrlProperties/devtools",
    "referencedFiles": [],
    "type": "chunk",
    "viteMetadata": {
      "importedAssets": Set {},
      "importedCss": Set {},
    },
  },
  {
    "code": "import { l as log } from \\"../../../../log.js\\";
log(\\"options\\");
",
    "dynamicImports": [],
    "exports": [],
    "facadeModuleId": "vite-plugin-web-extension/test/manifest/resources/htmlUrlProperties/options.html",
    "fileName": "assets/test/manifest/resources/htmlUrlProperties/options.js",
    "implicitlyLoadedBefore": [],
    "importedBindings": {
      "assets/log.js": [
        "l",
      ],
    },
    "imports": [
      "assets/log.js",
    ],
    "isDynamicEntry": false,
    "isEntry": true,
    "isImplicitEntry": false,
    "map": null,
    "modules": {
      "vite-plugin-web-extension/test/manifest/resources/htmlUrlProperties/options.html": {
        "code": "",
        "originalLength": 226,
        "removedExports": [],
        "renderedExports": [],
        "renderedLength": 0,
      },
      "vite-plugin-web-extension/test/manifest/resources/htmlUrlProperties/options.js": {
        "code": "log(\\"options\\");",
        "originalLength": 52,
        "removedExports": [],
        "renderedExports": [],
        "renderedLength": 15,
      },
    },
    "name": "test/manifest/resources/htmlUrlProperties/options",
    "referencedFiles": [],
    "type": "chunk",
    "viteMetadata": {
      "importedAssets": Set {},
      "importedCss": Set {},
    },
  },
  {
    "code": "import { l as log } from \\"../../../../log.js\\";
log(\\"popup\\");
",
    "dynamicImports": [],
    "exports": [],
    "facadeModuleId": "vite-plugin-web-extension/test/manifest/resources/htmlUrlProperties/popup.html",
    "fileName": "assets/test/manifest/resources/htmlUrlProperties/popup.js",
    "implicitlyLoadedBefore": [],
    "importedBindings": {
      "assets/log.js": [
        "l",
      ],
    },
    "imports": [
      "assets/log.js",
    ],
    "isDynamicEntry": false,
    "isEntry": true,
    "isImplicitEntry": false,
    "map": null,
    "modules": {
      "vite-plugin-web-extension/test/manifest/resources/htmlUrlProperties/popup.html": {
        "code": "",
        "originalLength": 224,
        "removedExports": [],
        "renderedExports": [],
        "renderedLength": 0,
      },
      "vite-plugin-web-extension/test/manifest/resources/htmlUrlProperties/popup.js": {
        "code": "log(\\"popup\\");",
        "originalLength": 50,
        "removedExports": [],
        "renderedExports": [],
        "renderedLength": 13,
      },
    },
    "name": "test/manifest/resources/htmlUrlProperties/popup",
    "referencedFiles": [],
    "type": "chunk",
    "viteMetadata": {
      "importedAssets": Set {},
      "importedCss": Set {},
    },
  },
  {
    "fileName": "manifest.json",
    "name": undefined,
    "source": "{
  \\"version\\": \\"1.0.0\\",
  \\"name\\": \\"Manifest Name\\",
  \\"manifest_version\\": 2,
  \\"browser_action\\": {
    \\"default_popup\\": \\"test/manifest/resources/htmlUrlProperties/popup.html#hashValue\\"
  },
  \\"devtools_page\\": \\"test/manifest/resources/htmlUrlProperties/devtools.html?query=1\\",
  \\"options_ui\\": {
    \\"page\\": \\"test/manifest/resources/htmlUrlProperties/options.html?query=1#hashValue\\"
  }
}",
    "type": "asset",
  },
  {
    "fileName": "test/manifest/resources/htmlUrlProperties/devtools.html",
    "name": undefined,
    "source": "<!DOCTYPE html>
<html lang=\\"en\\">
  <head>
    <meta charset=\\"UTF-8\\" />

    <script type=\\"module\\" src=\\"http://example.com/httpModuleScript.js\\"></script>
    <script type=\\"module\\" crossorigin src=\\"/assets/test/manifest/resources/htmlUrlProperties/devtools.js\\"></script>
  </head>
</html>
",
    "type": "asset",
  },
  {
    "fileName": "test/manifest/resources/htmlUrlProperties/options.html",
    "name": undefined,
    "source": "<!DOCTYPE html>
<html lang=\\"en\\">
  <head>
    <meta charset=\\"UTF-8\\" />

    <script type=\\"module\\" src=\\"http://example.com/httpModuleScript.js\\"></script>
    <script type=\\"module\\" crossorigin src=\\"/assets/test/manifest/resources/htmlUrlProperties/options.js\\"></script>
  </head>
</html>
",
    "type": "asset",
  },
  {
    "fileName": "test/manifest/resources/htmlUrlProperties/popup.html",
    "name": undefined,
    "source": "<!DOCTYPE html>
<html lang=\\"en\\">
  <head>
    <meta charset=\\"UTF-8\\" />

    <script type=\\"module\\" src=\\"http://example.com/httpModuleScript.js\\"></script>
    <script type=\\"module\\" crossorigin src=\\"/assets/test/manifest/resources/htmlUrlProperties/popup.js\\"></script>
  </head>
</html>
",
    "type": "asset",
  },
]
`;
